app/models/unit.rb

```diff
@@ -92,13 +92,49 @@ def role_for(user)
   has_many :convenors, -> { joins(:role).where('roles.name = :role', role: 'Convenor') }, class_name: 'UnitRole'
   has_many :staff, ->     { joins(:role).where('roles.name = :role_convenor or roles.name = :role_tutor', role_convenor: 'Convenor', role_tutor: 'Tutor') }, class_name: 'UnitRole'
 
+  # Unit has a teaching period
+  belongs_to :teaching_period
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+
+  validate :validate_end_date_after_start_date
+  validate :ensure_teaching_period_dates_match, if: :has_teaching_period?
```

Comment:
Can we not override start_date and end_date getters so that they access the teaching period value if it is there?

Reply:
I added that, but you asked me to remove it because it was redundant. If there is a teaching period, then I am manually writing the attribute value, so it is getting the teaching period value if it is there.
If you want, we can still add it.

Reply:
Ok - lets leave this as is now.